packages/rest/src/lib/CDN.ts

```diff
@@ -0,0 +1,147 @@
+const ALLOWED_EXTENSIONS = ['webp', 'png', 'jpg', 'jpeg', 'gif'] as const;
+const ALLOWED_SIZES = [16, 32, 64, 128, 256, 512, 1024, 2048, 4096] as const;
+
+export type ImageExtension = typeof ALLOWED_EXTENSIONS[number];
+export type ImageSize = typeof ALLOWED_SIZES[number];
+
+export interface ImageURLOptions {
+	extension?: ImageExtension;
+	size?: ImageSize;
+	dynamic?: boolean;
+}
+
+/**
+ * The CDN link builder
+ */
+export class CDN {
+	public constructor(private readonly base: string) {}
```

Comment:
```suggestion
import { DefaultRestOptions } from './utils/constants';

const ALLOWED_EXTENSIONS = ['webp', 'png', 'jpg', 'jpeg', 'gif'] as const;
const ALLOWED_SIZES = [16, 32, 64, 128, 256, 512, 1024, 2048, 4096] as const;

export type ImageExtension = typeof ALLOWED_EXTENSIONS[number];
export type ImageSize = typeof ALLOWED_SIZES[number];

export interface ImageURLOptions {
	extension?: ImageExtension;
	size?: ImageSize;
	dynamic?: boolean;
}

/**
 * The CDN link builder
 */
export class CDN {
	public constructor(private readonly base: string = DefaultRestOptions.cdn) {}
```

Reply:
`ALLOWED_EXTENSIONS` and `ALLOWED_SIZES` should be `Set`s for faster validation (L135 and L139).

Reply:
They're not faster for small datasets, hashing a value takes a lot more time than iterating over 10 entries.